lib/jnpr/junos/factcache.py

```diff
@@ -0,0 +1,280 @@
+from abc import ABCMeta
+import collections
+import warnings
+
+import jnpr.junos.facts
+import jnpr.junos.exception
+
+
+class _FactCache(collections.MutableMapping):
```

Comment:
Now we get

type(dev.facts)
<class 'jnpr.junos.factcache._FactCache'>

Hope user is not comparing it with dict anywhere